primitives/src/types.rs

```diff
@@ -64,12 +64,17 @@ pub type Timestamp = u64;
 /// Block header type as expected by this runtime.
 pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
 
+/// Block type.
+pub type Block = generic::Block<Header, OpaqueExtrinsic>;
+
+/// Block ID.
+pub type BlockId = generic::BlockId<Block>;
+
 /// The outer `FeeExchange` type. It is versioned to provide flexibility for future iterations
 /// while maintaining backward compatibility.
 #[derive(PartialEq, Eq, Clone, Encode, Decode, Debug)]
 pub enum FeeExchange<AssetId, Balance> {
 	/// A V1 FeeExchange
-	#[codec(compact)]
```

Comment:
why remove this?

Reply:
It is removed as part of fixing cennzx build issues. I need to put it back and chack what was the actual build issue.

Reply:
`error: Invalid attribute on variant, only `#[codec(skip)]` and `#[codec(index = $u8)]` are accepted.`

Reply:
Ok I think it's safe to ignore, seems like `enum` doesn't support 'compact' in codec 2.0